shell/platform/fuchsia/flutter/engine.cc

```diff
@@ -248,15 +251,23 @@ Engine::Engine(Delegate& delegate,
       auto compositor_context =
           std::make_unique<flutter_runner::CompositorContext>(
               session_connection_.value(), surface_producer_.value(),
-              legacy_external_view_embedder_);
+              legacy_external_view_embedder_.value());
+
       return std::make_unique<flutter::Rasterizer>(
           shell, std::move(compositor_context));
     } else {
       return std::make_unique<flutter::Rasterizer>(shell);
     }
   };
 #else
-  on_create_rasterizer = [](flutter::Shell& shell) {
+  on_create_rasterizer = [this](flutter::Shell& shell) {
+    FML_DCHECK(surface_producer_);
+
+    WarmupSkps(
```

Comment:
I highly recommend wrapping all of the logic you're adding in a config flag

Otherwise a revert will make you very sad :(